apps/sms/services/js/moz_mobile_message/moz_mobile_message_shim.js

```diff
@@ -71,6 +71,22 @@ function cloneThread(thread) {
   };
 }
 
+function clone(objectToBeCloned) {
+  // Basis checking.
+  if (!(objectToBeCloned instanceof Object)) {
+    return objectToBeCloned;
+  }
+
+  var objectClone = {};
+
+  // Clone each property.
+  for (var prop in objectToBeCloned) {
```

Comment:
actually I'm not sure this works; I think properties for messages are not enumerable :/ (but please try it, maybe it is now)